torch/csrc/jit/ir/alias_analysis.cpp

```diff
@@ -1409,38 +1467,21 @@ c10::optional<Element*> AliasDb::setWildcard(const Value* v) {
   if (!maybe_wildcardElement) {
     return c10::nullopt;
   }
+  wildcards_.insert(v);
+  auto e = getOrCreateElement(v);
+  wc_[e].push_back(v);
 
-  auto wildcardElement = *maybe_wildcardElement;
-  // Making a value a wildcard means that all its potential memory locations
-  // may alias the wildcard set.
-  const MemoryLocations pointeeSet =
-      getOrCreateElement(v)->getMemoryLocations();
-  for (const auto& pointee : pointeeSet) {
-    auto from = memoryDAG_->fromIndex(pointee);
-    // avoid cycles where the wildcard points to itself
-    if (from != wildcardElement) {
-      memoryDAG_->makePointerTo(from, wildcardElement);
-    }
-  }
-
-  // We also need to update the write index with new writes to the wildcard set.
-  for (auto& pr : writeIndex_) {
-    auto& writtenTo = pr.second;
-    if (writtenTo.intersects(pointeeSet)) {
-      writtenTo.set(wildcardElement->index);
-    }
-  }
-  isWriteCacheStale_ = true;
-  return wildcardElement;
+  return *maybe_wildcardElement;
 }
 
-void AliasDb::rebuildWriteCache() const {
-  writeCache_ = {};
-  for (const auto& pr : writeIndex_) {
+MemoryLocations AliasDb::buildWriteCache() const {
+  MemoryLocations ret;
```

Comment:
Why was this changed from mutating & void to functional and returning a new value?

Reply:
Just because it is easier for me to reason about functional things